elements/nuxeo-document-bulk-actions/nuxeo-edit-documents-button.js

```diff
@@ -0,0 +1,501 @@
+       * until a response is received from the server.
+       */
+      saving: {
+        type: Boolean,
+        value: false,
+        readOnly: true,
+      },
+      /**
+       * The path of the layout to be loaded.
+       */
+      _href: {
+        type: String,
+        readOnly: true,
+      },
+      /**
+       * A representation of the properties being edited.
+       * It should closely resemble the `document` object present in the loaded bulk layout, except for a few details:
+       * - Contains the full bound paths to the properties of the layout (`document.properties.[schema].[field]`),
+       *   which is useful for obtaining the bulk widget wrapper instances to which they belong.
+       * - When the layout is loaded, this gets populated with all the bound paths that can be obtained through
+       *   introspection (different from `document`, which will only add paths as the fields are being edited).
+       */
+      _model: Object,
+    };
+  }
+
+  static get observers() {
+    return ['_loadLayout(documents.splices, layout, hrefFunction, hrefBase)'];
+  }
+
+  constructor() {
+    super();
+    this.icon = 'nuxeo:edit';
+    this.label = this.i18n('documentEditBulkAction.tooltip');
+    this.operation = 'Document.Update';
+    // set up a listener to act on changes of the update modes of the bulk widget wrappers in the layout
+    this.addEventListener('update-mode-changed', (e) => this._updateBulkWidget(e.detail));
+  }
+
+  /**
+   * Control the visibility of the button.
+   */
+  _isHidden() {
+    // TODO compare to the _isHidden on nuxeo-add-to-collections-documents-button
+    return !(isPageProviderDisplayBehavior(this.documents) || (this.documents && this.documents.length));
+  }
+
+  /**
+   * Checks for the validity of the form, taking into account the specificities of the bulk widget wrappers.
+   */
+  _validate() {
+    let valid = true;
+    const layout = this.$$('nuxeo-layout');
+    if (layout.element) {
+      const elements = layout._getValidatableElements(layout.element.root);
+      for (let el, i = 0; i < elements.length; i++) {
+        el = elements[i];
+        if (el.tagName.toLowerCase() === 'nuxeo-bulk-widget') {
+          // validate the element as bulk widget
+          // if the bulk widget is not tagged with a `boundPath`, it's safe to assume it has no value
+          const value = el.boundPath ? this._model[el.boundPath] : null;
+          // check if trying to replace with empty value
+          const replacingWithEmptyValue = el.updateMode === 'replace' && this._isValueEmpty(value);
+          if (replacingWithEmptyValue) {
+            el._setError(this.i18n('bulkWidget.error.replaceWithEmpty'));
+          }
+          valid = !replacingWithEmptyValue && valid;
+        } else {
+          // validate the element as any other widget
+          valid = (el.validate ? el.validate() : el.checkValidity()) && valid;
```

Comment:
nit:
```suggestion
          valid &&= el.validate ? el.validate() : el.checkValidity();
```
(assuming it works with linting)

Reply:
```
error [parse-error] - This experimental syntax requires enabling the parser plugin: 'logicalAssignment'
```
🤷